browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -50,12 +55,91 @@ bool ApplyPotentialReferrerBlock(std::shared_ptr<BraveRequestInfo> ctx) {
   return false;
 }
 
+bool ApplyPotentialQueryStringFilter(std::shared_ptr<BraveRequestInfo> ctx) {
```

Comment:
I'd prefer this function to take only URL and the resulted URL, not the whole context

Reply:
That would involve moving a lot of the logic for whether or not to touch the query string directly into [`OnBeforeURLRequest_SiteHacksWork()`](https://github.com/brave/brave-core/blob/e9b80b829aded2c6cb615c0ac2e18faf1b1cb457/browser/net/brave_site_hacks_network_delegate_helper.cc#L55-L60) and making it look very different from `ApplyPotentialReferrerBlock()`[https://github.com/brave/brave-core/blob/e9b80b829aded2c6cb615c0ac2e18faf1b1cb457/browser/net/brave_site_hacks_network_delegate_helper.cc#L32].

Reply:
hmm, which logic? You can just pass an URL and a pointer to `new_url_spec`, and nothing changes in `OnBeforeURLRequest_SiteHacksWork()`. As for `ApplyPotentialReferrerBlock()` I think that some day it will be refactored to look like this new function :) Anyways, this is a minor.

Reply:
Ah I see what you mean. Yes, I think I can make that work nicely with a pointer to `new_url_spec`.